bika/lims/workflow/sample/events.py

```diff
@@ -115,30 +112,37 @@ def after_sample(obj):
     """Method triggered after a 'sample' transition for the Sample passed in.
     Triggers the 'sample' transition for dependent objects, such as Sample
     Partitions and Analysis Requests.
+
+    Sampling workflow is complete, sample will be put in one of these states:
+      - to_be_preserved
+      - sample_due
+      - received
+
     This function is called automatically by
     bika.lims.workflow.AfterTransitionEventHandler
     :param obj: Sample affected by the transition
     :type obj: Sample
     """
     _cascade_transition(obj, 'sample')
 
+    def next_transition(o):
+        auto = obj.bika_setup.getAutoReceiveSamples()
+        if auto and isTransitionAllowed(o, 'receive'):
+            return 'receive'
+        else:
+            return 'sample_due'
+
+    lowest_state_transition = next_transition(obj)
```

Comment:
You don't need all this logic if you move `autoReceiveSamples()` condition to the guard. This code can be simplified like follows then:
```python
lowest_state_transition = isTransitionAllowed(obj, 'receive') and 'receive' or 'sample_due'
```